feat(boards): add siyi_unifc-6-pico flight controller (#27835)
PX4/PX4-Autopilot: docs/en/flight_controller/siyi-unifc-6-pico.md

# SIYI UniFC 6 PICO

> <Badge type="tip" text="PX4 v1.18" />

Reviewer: @mrpollo Is entry to v1.18 now blocked?

```suggestion
<Badge type="tip" text="PX4 main (v1.19)" />
```

Reviewer 2: The next release is 2.0

> <Badge type="tip" text="PX4 main (v1.19)" />

Reviewer: ```suggestion
<Badge type="tip" text="PX4 main (v2.0)" />
```

<img src="../../assets/flight_controller/siyi-unifc-6-pico/siyi-unifc-6-pico-top.png" />

::: info
This flight controller is [manufacturer supported](../flight_controller/autopilot_manufacturer_supported.md).
:::

## Key Features

- **MCU:** STM32H743II (32 Bit Arm® Cortex®-M7, 480 MHz, 2MB Flash, 1MB RAM)
- **IMU (Dual):** TDK InvenSense ICM-45686 (primary, SPI4) + Bosch BMI088 accel/gyro (secondary, SPI1)
- **Barometer (Dual):** InvenSense ICP201XX (internal at address 0x63, external at address 0x64)
- **Magnetometer:** iSentek IST8310
- **Interfaces:**
  - 6x UARTs (including GPS, telemetry, and RC ports)
  - 2x CAN (DroneCAN / UAVCAN)
  - 4x I²C (2 external, 2 internal)
  - 1x dedicated RC Input (SBUS / DSM / SUMD)
  - 14x PWM outputs (DShot-capable)
  - SDMMC1 4-bit Micro-SD slot (high-speed logging)
  - Ethernet (RMII, LAN8742A PHY)
  - USB
  - Onboard safety button
- **Power:**
  - Single analog battery voltage/current channel (BAT1)
  - Voltage sensing up to 14S LiPo by default
  - On-board BECs for peripheral power (consult manufacturer for exact rail specifications)

## Where to Buy

> Order from [SIYI](https://siyi.biz/).

Reviewer: I don't see this product on your site

Author: This flight controller is expected to be available in September.

## Physical / Mechanical

> Refer to the manufacturer's datasheet for the latest mechanical drawing, mounting pattern, and weight specifications.

Reviewer: LInk to sheet please

## Specifications

### Processors & Sensors

- **FMU Processor:** STM32H743II
  - 32 Bit Arm® Cortex®-M7, 480 MHz
  - 2MB Flash, 1MB RAM
- **On-board Sensors:**
  - Accel/Gyro 1: TDK InvenSense ICM-45686 (SPI4)
  - Accel/Gyro 2: Bosch BMI088 (SPI1)
  - Barometer 1: InvenSense ICP201XX (I²C3, address `0x63`)
  - Barometer 2: InvenSense ICP201XX (I²C2, address `0x64`)
  - Compass: iSentek IST8310 (I²C3, address `0x0E`)

### Power Configuration

The board provides **one analog battery monitoring channel**:

- **BAT1** - Battery voltage and current sensed on `ADC1_INP6` (voltage, PF12) and `ADC1_INP2` (current, PF11).

The channel feeds PX4's standard analog battery driver with a default 18.181 voltage divider and 36.366 A/V current ratio, supporting up to 14S LiPo batteries out of the box.

## Connectors & Pinouts

> The board exposes the following connectors and solder pads:

Reviewer: What ports connectors? Are they Pixhawk standard https://github.com/pixhawk/Pixhawk-Standards

If not, what are they? The reason it matters is that it affects whether you're likely to be able to just plug in standard components such as GPS from other vendors. 

I can see GPS1 is probably the full-GPS connector below.

Author: I forgot to reference the pinout image. Thanks for reminding me.

Only the receive line (`RX8`) is broken out, so the port supports:

- Single-wire SBUS (inverted)
- Spektrum DSM
- Graupner SUMD

Configure the receiver protocol with the relevant `RC_*_PRT_CFG` parameter from QGroundControl, pointing it at the `RC` slot.

### PWM Output Groups

The 14 PWM outputs are organized into four timer groups. Channels in the same group must use the same output rate and DShot mode:

| Group | Timer | Channels |
| ----- | ----- | -------- |
| 1     | TIM1  | M1–M4    |
| 2     | TIM4  | M5–M8    |
| 3     | TIM5  | M9–M12   |
| 4     | TIM12 | M13–M14  |

Channels within the same group need to use the same output rate.
If any channel in a group uses DShot then all channels in the group need to use DShot.

### Ethernet

The board features an Ethernet port with RMII interface (LAN8742A PHY).
Only the four RMII data channels are broken out to external pins: RMII_RXD0, RMII_RXD1, RMII_TXD0, RMII_TXD1.
Other RMII signals (REF_CLK, CRS_DV, MDC, MDIO) are connected internally to the onboard PHY chip.
MAVLink is pre-configured on Ethernet interface (MAV_2_CONFIG set to 1000).

## Building / Loading Firmware

> ::: tip
> Most users will not need to build this firmware (from PX4 v1.18).
> It is pre-built and automatically installed by _QGroundControl_ when appropriate hardware is connected.
> :::

Reviewer: ```suggestion
::: tip
Most users will not need to build this firmware (from PX4 v2.0).
It is pre-built and automatically installed by _QGroundControl_ when appropriate hardware is connected.
:::
```

To [build PX4](../dev_setup/building_px4.md) for this target from source:

```sh
make siyi_unifc-6-pico_default
```

Initial firmware flashing can be done over USB via QGroundControl.

> The bootloader status follows the standard generic PX4 LED indications (Red = Bootloader / Error, Blue = Active / Activity, Green = Powered).

Reviewer: src/hw_config.h defines BOARD_PIN_LED_BOOTLOADER = GPIO_nLED_GREEN and BOARD_PIN_LED_ACTIVITY = GPIO_nLED_BLUE. bootloader is Green in this board's own source, not Red. Looks like unedited boilerplate copied from another board's page.

| Group | Timer | Channels |
| ----- | ----- | -------- |
| 1     | TIM1  | M1–M4    |
| 2     | TIM4  | M5–M8    |
| 3     | TIM5  | M9–M12   |
| 4     | TIM12 | M13–M14  |

Channels within the same group need to use the same output rate.
If any channel in a group uses DShot then all channels in the group need to use DShot.

### Ethernet

The board features an Ethernet port with RMII interface (LAN8742A PHY).
Only the four RMII data channels are broken out to external pins: RMII_RXD0, RMII_RXD1, RMII_TXD0, RMII_TXD1.
Other RMII signals (REF_CLK, CRS_DV, MDC, MDIO) are connected internally to the onboard PHY chip.
MAVLink is pre-configured on Ethernet interface (MAV_2_CONFIG set to 1000).

## Building / Loading Firmware

::: tip
Most users will not need to build this firmware (from PX4 v2.0).
It is pre-built and automatically installed by _QGroundControl_ when appropriate hardware is connected.
:::

To [build PX4](../dev_setup/building_px4.md) for this target from source:

```sh
make siyi_unifc-6-pico_default
```

Initial firmware flashing can be done over USB via QGroundControl.

> The bootloader status don't follows the standard generic PX4 LED indications (Green = Bootloader / Error, Blue = Active / Activity, Red = Powered).

Reviewer 3: "does not follow"

Author: I have  corrected it.
